test.js

```diff
@@ -447,3 +451,50 @@ test('alias - require previousId and userId', t => {
     })
   })
 })
+
+test('e2e', async t => {
+  const id = uid(16)
+
+  // Segment Write Key for https://segment.com/segment-libraries/sources/analytics_node_e2e_test/overview
+  const analytics = new Analytics('wZqHyttfRO0KxEHyRTujWZQswgTDZx1N')
+  analytics.track({
+    userId: 'prateek',
+    event: 'E2E Test',
+    properties: {
+      id
+    }
+  })
+  analytics.flush()
+
+  // Give some time for events to be delivered from the API to destinations.
+  await sleep(5000)
+
+  const axiosClient = axios.create({
+    baseURL: 'https://api.runscope.com',
+    timeout: 10 * 1000,
+    headers: { 'Authorization': `Bearer ${process.env.RUNSCOPE_TOKEN}` }
+  })
+  retries(axiosClient, { retries: 3 })
+
+  // Runscope Bucket for https://www.runscope.com/stream/zfte7jmy76oz.
+  const messagesResponse = await axiosClient.get('buckets/zfte7jmy76oz/messages?count=10')
+  t.is(messagesResponse.status, 200)
+
+  let found = false
+
+  messagesResponse.data.data.forEach(async item => {
+    const response = await axiosClient.get('buckets/zfte7jmy76oz/messages/' + item.uuid)
+    t.is(response.status, 200)
+
+    const body = JSON.parse(response.data.data.request.body)
```

Comment:
I think Axios will return JSON for us if we ask it to.

Reply:
How do I get it to do that? I tried:

```
    const response = await axiosClient.get(`buckets/zfte7jmy76oz/messages/${item.uuid}`, {
      responseType: 'json'
    })
```

Reply:
Try passing it in the options when creating `axiosClient`.

Reply:
Still failed.

```
const axiosClient = axios.create({
      baseURL: 'https://api.runscope.com',
      timeout: 10 * 1000,
      headers: { Authorization: `Bearer ${process.env.RUNSCOPE_TOKEN}` },
      responseType: 'json'
    })
```

```
for (const item of messagesResponse.data.data) {
      const response = await axiosClient.get(`buckets/zfte7jmy76oz/messages/${item.uuid}`)
      t.is(response.status, 200)

      if (id === response.data.data.request.body.properties.id) {
        found = true
        break
      }
    }
```

```
> analytics-node@3.1.1 test /Users/prateek/dev/src/github.com/segmentio/analytics-node
> standard && ava


  30 passed
  1 failed

  end to end test
  /Users/prateek/dev/src/github.com/segmentio/analytics-node/test.js:491

   490:                                                                  
   491:       if (id === response.data.data.request.body.properties.id) {
   492:         found = true                                             

  Rejected promise returned by test. Reason:

  TypeError {
    message: 'Cannot read property \'id\' of undefined',
  }

  Test.<anonymous> (test.js:491:18)
  step (test.js:52:191)
npm ERR! Test failed.  See above for more details.

```

Reply:
Error is different here, `properties` is undefined.